From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Fri, 1 Nov 2019 16:58:55 +0100
Subject: [PATCH] allow building without NEON

The build-system currently assumes that NEON code can always be built on
ARM. Make it possible to disable this via the 'make' command-line.

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 nss/lib/freebl/Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/nss/lib/freebl/Makefile b/nss/lib/freebl/Makefile
index 5943fb377894..1401f8154fcc 100644
--- a/nss/lib/freebl/Makefile
+++ b/nss/lib/freebl/Makefile
@@ -124,6 +124,7 @@ ifeq ($(CPU_ARCH),aarch64)
     EXTRA_SRCS += aes-armv8.c gcm-aarch64.c
 endif
 ifeq ($(CPU_ARCH),arm)
+  ifndef NO_NEON
     ifdef CC_IS_CLANG
         DEFINES += -DUSE_HW_AES
         EXTRA_SRCS += aes-armv8.c
@@ -138,6 +139,7 @@ ifeq ($(CPU_ARCH),arm)
             EXTRA_SRCS += aes-armv8.c
         endif
     endif
+  endif
 endif
 
 ifeq ($(OS_TARGET),OSF1)
